eventmesh-sdks/eventmesh-sdk-java/src/main/java/org/apache/eventmesh/client/tcp/common/TcpClient.java

```diff
@@ -181,10 +181,12 @@ protected void send(Package msg) throws Exception {
     protected Package io(Package msg, long timeout) throws Exception {
         Object key = RequestContext.key(msg);
         RequestContext context = RequestContext.context(key, msg);
-        if (!contexts.containsValue(context)) {
-            contexts.put(key, context);
-        } else {
-            LogUtils.info(log, "duplicate key : {}", key);
+        synchronized (context) {
+            if (!contexts.containsValue(context)) {
+                contexts.put(key, context);
+            } else {
+                LogUtils.info(log, "duplicate key : {}", key);
+            }
```

Comment:
When the client sends two identical messages due to network issues, the existing logic can avoid duplicate keys. However, if the client sends two messages with the same content but different `seq` due to business issues, the existing logic, when running in a thread-unsafe manner, will retain the key and message from the first received package and will not be able to output a duplicate key log. Therefore, it is still necessary to use "putIfAbsent."

```suggestion
RequestContext previousContext = contexts.putIfAbsent(key, context);
if (previousContext != null) {
    LogUtils.info(log, "duplicate key : {}", key);
}
```